src/audio/pipeline/pipeline-schedule.c

```diff
@@ -46,21 +46,33 @@ static enum task_state pipeline_task_cmd(struct pipeline *p,
 	int err, cmd = p->trigger.cmd;
 
 	if (!p->trigger.host) {
+		int ret;
+
 		p->trigger.cmd = COMP_TRIGGER_NO_ACTION;
 
 		switch (cmd) {
 		case COMP_TRIGGER_STOP:
 		case COMP_TRIGGER_PAUSE:
-			return p->trigger.aborted ? SOF_TASK_STATE_RUNNING :
-				SOF_TASK_STATE_COMPLETED;
+			if (p->trigger.aborted)
+				ret = SOF_TASK_STATE_RUNNING;
+			else
+				ret = SOF_TASK_STATE_COMPLETED;
```

Comment:
(to the first commit) you could keep it as
```
ret = p->trigger.aborted ? SOF_TASK_STATE_RUNNING : SOF_TASK_STATE_COMPLETED;
```

Reply:
sure, then it would be
```
ret = p->trigger.pending || p->trigger.aborted ? SOF_TASK_STATE_RUNNING : SOF_TASK_STATE_COMPLETED;
```

Reply:
@lyakh I personally prefer the if/else instead of ?:. I think it makes the code more readable.